kibana-reports/server/backend/opendistro-es-reports-plugin.ts

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import { ES_REPORTS_API } from '../../common';
+
+export default function (Client: any, config: any, components: any) {
+  const clientAction = components.clientAction.factory;
+
+  Client.prototype.es_reports = components.clientAction.namespaceFactory();
+  const esReports = Client.prototype.es_reports.prototype;
+
+  /**
+   * report related APIs
+   */
+  esReports.createReport = clientAction({
+    url: {
+      fmt: `${ES_REPORTS_API.ON_DEMAND_REPORT}`,
+    },
+    method: 'PUT',
+    needBody: true,
+  });
+
+  esReports.updateReportInstanceStatus = clientAction({
+    url: {
+      fmt: `${ES_REPORTS_API.REPORT_INSTANCE}?id=<%=reportId%>`,
+      req: {
+        reportId: {
+          type: 'string',
+          required: true,
+        },
+      },
+    },
+    method: 'POST',
+    needBody: true,
+  });
+
+  esReports.getReportById = clientAction({
+    url: {
+      fmt: `${ES_REPORTS_API.REPORT_INSTANCE}?id=<%=reportId%>`,
+      req: {
+        reportId: {
+          type: 'string',
+          required: true,
+        },
+      },
+    },
+    method: 'GET',
+  });
+
+  esReports.getReports = clientAction({
+    url: {
+      fmt: `${ES_REPORTS_API.LIST_REPORT_INSTANCES}`,
+      //TODO: wrong format error thrown even required = false, need to figure it out the correct setting to make it truly optional
+      // req: {
```

Comment:
if we have the `TODO` do we also need to still leave in the commented-out code?

Reply:
This is one of the todos I will address in future PRs